cpp/include/raft/neighbors/ivf_flat.cuh

```diff
@@ -397,37 +397,35 @@ void search(raft::device_resources const& handle,
  * [n_queries, k]
  * @param[out] distances a device pointer to the distances to the selected neighbors [n_queries, k]
  * @param[in] params configure the search
- * @param[in] k the number of neighbors to find for each query.
  */
-template <typename value_t, typename idx_t, typename int_t>
+template <typename value_t, typename idx_t>
 void search(raft::device_resources const& handle,
             const index<value_t, idx_t>& index,
             raft::device_matrix_view<const value_t, idx_t, row_major> queries,
             raft::device_matrix_view<idx_t, idx_t, row_major> neighbors,
             raft::device_matrix_view<float, idx_t, row_major> distances,
-            const search_params& params,
-            int_t k)
+            const search_params& params)
```

Comment:
I would prefer that we keep any params structs to be the second argument (after the handle) and then any additional single parameter arguments come last. 

This is the way it's done across the codebase. Kmeans is a good example of an API outside of neighbors that uses that design.